packages/qwik/src/core/api.md

```diff
@@ -681,6 +686,12 @@ export interface QuoteHTMLAttributes<T extends Element> extends Attrs<'q', T> {
 // @public @deprecated (undocumented)
 export type QwikAnimationEvent<T = Element> = NativeAnimationEvent;
 
+// @public
+export interface QwikAttributes<EL extends Element> extends QwikAttributesBase, RefAttr<EL>, QwikEvents<EL, false> {
+    // (undocumented)
+    class?: ClassList | undefined;

```

Comment:
Hey @maiieul @wmertens, is there any reason why `Signal<ClassList>` wasn't included here or should it be part of it? I'm not 100% sure what the original issue is related to and if it was an explicit decision to remove this here or not.

My current use case of using e.g. `const ComponentProps = QwikIntrinsicElements<'img'> & {...additional stuff...}` to create an image-like now sees this definition over the one with `Signal<ClassList>` from `DOMAttributes`.

Happy to put up a PR addressing this if unintended.

Reply:
Yes the signal is deprecated actually, because it makes it very hard to forward a class by doing `class={[props.class, "hi"]}`.

Instead, you should just pass classSignal.value and the optimizer will forward it for you.